test/pylib/manager_client.py

```diff
@@ -513,14 +513,25 @@ async def rebuild_node(self, server_id: ServerNum,
                                    timeout=timeout)
         self._driver_update()
 
-    async def server_get_config(self, server_id: ServerNum) -> dict[str, object]:
+    async def server_get_config(self, server_id: ServerNum) -> dict[str, Any]:
         data = await self.client.get_json(f"/cluster/server/{server_id}/get_config")
         assert isinstance(data, dict), f"server_get_config: got {type(data)} expected dict"
         return data
 
-    async def server_update_config(self, server_id: ServerNum, key: str, value: object) -> None:
-        await self.client.put_json(f"/cluster/server/{server_id}/update_config",
-                                   {"key": key, "value": value})
+    async def server_update_config(self,
+                                   server_id: ServerNum,
+                                   key: str | None = None,
+                                   value: Any = None,
+                                   *,
+                                   values: dict[str, Any] | None = None) -> None:
+        if key is not None:
+            assert value is not None, "`value` is required if `key` is not None"
+            assert values is None, "`key: value` pair and `values` dict can't be used simultaneously"
+            values = {key: value}
+        else:
+            assert isinstance(values, dict)
+
+        await self.client.put_json(resource_uri=f"/cluster/server/{server_id}/update_config", data={"values": values})
```

Comment:
Name values misleading. It looks like you're passing only values of the dict rather than the full dict. Probably better naming would be `config` or `config_options`.

Reply:
changed to `config_options`